match/lib/match/encryption/openssl.rb

```diff
@@ -50,7 +50,8 @@ def decrypt_files
             UI.error("Couldn't decrypt the repo, please make sure you enter the right password!")
             UI.user_error!("Invalid password passed via 'MATCH_PASSWORD'") if ENV["MATCH_PASSWORD"]
             clear_password
-            self.decrypt_files # call itself
+            # After clearing the password from Keychain, retry if MATCH_PASSWORD is set
+            self.decrypt_files if ENV["MATCH_PASSWORD"]
```

Comment:
This _does_ seem like it would work here but I think catching it here is more about an incorrect password 🤔 

I think we want instead want to see fetch the password outside of begin/rescue so the `UI.user_error!` when in a non-interactive doesn’t get caught here

1. I would also renamed `def password` to `def fetch_password!` here - https://github.com/fastlane/fastlane/blob/master/match/lib/match/encryption/openssl.rb#L86-L105
  - The exclamation point being important since this can throw when in non-interative
2. At the start of the `decrypt_files` method, we would then to `password = fetch_password!` so that I can throw if non-interactive before getting into the begin/rescue and causing recursion 

Let me know if you have any questions!

Reply:
Great feedback! I'll pick this up again when I can! 🙏

Reply:
Addressed @joshdholtz , let me know if that's what you pictured 😊